sherlock/document.py

```diff
@@ -180,6 +239,32 @@ def from_dict(cls, doc: "Document", dct: Dict[str, Any]) -> "Entity":
 
 @dataclass(frozen=True)
 class Relation:
+    """
+    Representation for a certain Relation within a Document.
+
+    A Relation is a specific instantiation of a relationship between
+    two Mentioned Entities within the Document.
+    Saves a specific Relation with its label, its head and tail Entity.
+    TODO: this is not consistent with Entity:
```

Comment:
Relation is between Mentions, not Entities, so it refers to a concrete text span (same as Mention). An Entity is the reference to the concrete concept, and does not need a reference to a text span (mention). Similarly, we could have an abstract type for a Relation (and rename the original "Relation" to "RelationMention"), where a Relation refers to a list of RelationMentions in text, and the Relation class has reference ids to concrete KBs, e.g. Wikidata (if those KBs actually have separate IDs for a concrete triple like 'Angela Merkel', 'country_of_citizenship', 'Germany' (https://www.wikidata.org/wiki/Q567 -> Property P27. Probably they don't, so may refids is not useful in such a case)

Reply:
Do I understand correctly that Relations do not refer to text spans directly, but to Mentions which in turn refer to text spans?

And your example for Relation, RelationMention would be the solution for the TODO written. I am not sure if it is worth to change (it adds "unecessary" code), but it would make it consistent with the approach with Entities + add your mentioned KB-connection benefit.